### ISSUE TITLE
Changes to CoreBrainInternal inspector fields don't always persist 

### ISSUE THREAD

**Issue by kevincog-msft** [Unity-Technologies/ml-agents#602]
I've found that changes to the CoreBrainInternal fields in the inspector don't always persist, particularly after the first time you modify them. For example, this can happen when changing the node name fields. These steps consistently reproduce the issue for me:

1. Reset the brain component.
2. Set it to use the Internal brain type.
3. Change one of the text fields (e.g. Graph Scope) to a different value.
4. Save the project and close Unity.
5. Reopen the project and change the text field again.
6. Save the project, close Unity, and reopen it again.

After reopening, the field will show the value from step 3, not step 5. To successfully persist changes, I need to reset the component values and re-enter everything.

**Comment by superjayman** [comment]
I have also noticed this issue. I'm also having an issue it may be related, I've trained 4 separate brains in one session just like the Soccer Example(They have two brains here Goalie and Kicker), but when I import the byte file from my training I'm getting an error

The node BRAIN_ML0/action could not be found. Please make sure the graphScope BRAIN_ML0/ is correct

The thing is I've entered the correct scope name, and everything looks fine just like the soccer example but it does not work!!

**Comment by unityjeffrey** [comment]
thanks @kevincog-msft and @superjayman - we've logged this bug and will report back when we have a resolution,

**Comment by xiaomaogy** [comment]
@kevincog-msft I assume @vincentpierre has replied to you on this issue in your ml-agents email list, so I am closing this issue. Feel free to open if there is more to be discussed here. 

To explain what happened here for people who is not in the email list, this is something related to Unity Editor. The Unity won't save the change under this case, but if you change something in the scene after step 5 (like moving object order in the hierarchy window) then save, the change would be saved.

**Comment by xiaomaogy** [comment]
@superjayman I think the error you raised is a different issue. Please create a new issue and post the snapshot and your C# script in that new issue post for others to help you.

**Comment by awjuliani** [comment]
Hi all. We will be re-working the Brain system to replace brains as gameobjects with brains as scriptable objects. This should alleviate the issues described here. Due to inactivity, I will be closing this issue for now.
